### ISSUE TITLE
Fails in Configuring when OpenCL is present in the system

### ISSUE THREAD

**Issue by Prasanth-rp** [ImageMagick/ImageMagick#1283]
When I tried to build ImageMagick in my desktop, I could see the configuration fails with "checking for libltdl" if OpenCL libraries found in the system.

If I remove the OpenCL libraries from my system the configuration skips this ltdl check step.

Then what does --disable-opencl --without-ltdl do here? By passing this it should skip the steps for verification isn't it, whether the system has OpenCL or not?

Steps to Reproduce
Install Opencl in your system and configure ImageMagick

./configure \
		--without-fftw --without-perl --disable-opencl --without-ltdl --without-modules
-------------------------------------------------------------
checking for libltdl... 
checking ltdl.h usability... yes
checking ltdl.h presence... yes
checking for ltdl.h... yes
checking for lt_dlinit in -lltdl... no
checking if libltdl package is complete... no -- some components failed test
configure: error: in `ImageMagick-7.0.7-22':
configure: error: libltdl is required for modules and OpenCL builds
See `config.log' for more details

Remove the OpenCL in your system and configure ImageMagick

./configure \
		--without-fftw --without-perl --disable-opencl --without-ltdl --without-modules

It passes the configuration and gets build successfully.

ImageMagick-7.0.7-22
Ubuntu 16.04.1

**Comment by urban-warrior** [comment]
Download ImageMagick-7.0.8-11, the lastest release.  Does the problem you reported persist?  If so, let us know and we will revisit the issue.

**Comment by rprasanth** [comment]
Hi @urban-warrior 

Have you got any chance to work on this?

**Comment by alvaropp** [comment]
same here!

**Comment by cristobalmejia** [comment]
same here

**Comment by zdavis** [comment]
Also an issue for us.

**Comment by RickyNotaro** [comment]
Same here on 7.0.8-57

**Comment by Taoey** [comment]
I have solved this problem.My OS is centos6.5（verified and succeed）.
You need install some libs 
```
yum install libtool-ltdl-devel fftw3-devel djvulibre-devel
```
--- 
and if your OS is not centos , you can search 'libltdl' in  https://pkgs.org，and install libltdl (unverified)

**Comment by rprasanth** [comment]
@Taoey 

This doesn't resolve the issue, the main issue is with the configure script, where it doesn't take the value --without-ltdl 
If this --without-ltdl  was passed to configuration it should not check for libtool libraries.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by HuangZhenyang** [comment]
I got the same error. And there has been `libltdl` in my Kali. 
So I tried `apt-cache search  ltdl` and then installed `libltdl-dev` with `apt-get install libltdl-dev`. 
Finally, it works!

**Comment by bseeger** [comment]
The fix recommended by @HuangZhenyang worked for me as well.

**Comment by suconghou** [comment]
![image](https://user-images.githubusercontent.com/4580719/66992278-1e29ee00-f0fc-11e9-8290-65e1c681af97.png)

still build failed in 7.0.8-68

**Comment by ajoslin103** [comment]
100% for the fix recommended by @HuangZhenyang

**Comment by rprasanth** [comment]
> 100% for the fix recommended by @HuangZhenyang

No, it's not, That's  the main issue opened, then why the configure script has this "--without-ltdl", if we need to install libltdl-dev

**Comment by ajoslin103** [comment]
@rprasanth is 100% correct 

I was able to build the ImageMagick rpms by installing libltdl-dev -- but now the resulting packages have a dependency on libltdl

**Comment by scholarboss** [comment]
I met the same problem in my debain. I solve it with the bottom command.
`sudo apt-get install -y libltdl-dev`

**Comment by rprasanth** [comment]
is the addressed issue was fixed?

**Comment by IllyaMoskvin** [comment]
I got this error while installing via Homebrew. I managed to mess up my symlinks, somehow. This fixed it:

```
$ brew install libtool
Warning: libtool 2.4.6_2 is already installed, it's just not linked
You can use `brew link libtool` to link this version.

$ brew link libtool
Linking /usr/local/Cellar/libtool/2.4.6_2... 20 symlinks created
```

After that, running `brew install imagemagick` worked fine!

**Comment by exekutive** [comment]
imagemagick 7.1.1

config log:
```
...
|       #include <ltdl.h>
| 
| int
| main (void)
| {
| 
|       lt_dlinit();
| 
|   ;
|   return 0;
| }
| 
configure:29790: checking if libltdl package is complete
configure:29811: result: no
configure:29815: error: in `/Users/joe/stage/im':
configure:29817: error: libltdl is required for modules and OpenCL builds
...
```

libltdl is part of the `libtool` package, which I have installed.
that header file doesn't look like mine

**Comment by exekutive** [comment]
if I use `--disable-osx-universal-binary`
then I simply get:
`'ltdl.h' file not found`

**Comment by pixobe** [comment]
I am facing same issue on centos9 , how is this resolved?
